spec/ks/nested_type_param.kst

```diff
@@ -0,0 +1,7 @@
+id: nested_type_param
+data: fixed_struct.bin
+asserts:
+  - actual: seq_block.repeated_thing[0]
+    expected: -52947
+  - actual: seq_block.repeated_thing[1]
+    expected: 1262698832
```

Comment:
I don't really like asserting these fields (see https://github.com/kaitai-io/kaitai_struct_tests/commit/eaff6822f59a7be422a2e6fd381a6d579057ae08#commitcomment-38238619), and I think they do not even reflect the intention. The only thing that you want to check here is that the parameter `repeat_count` was successfully passed in the type `test::my_type`, isn't it?

So why you just don't do:

```suggestion
  - actual: seq_block.repeat_count
    expected: 2
```

Beauty in simplicity, isn't it? It's not hiding the intent, and you are just using the bare minimum of features. Because you don't want to check if the first and second item of a dynamically sized-array are equal to something, that's just an indirect consequence of `repeat_count` being greater than or equal to 2.

And for the record, if you will ever need to assert arrays, please check also the length of the array. Because checking just the individual elements would pass even if the `repeat_count` is greater than 2, but that's not what we want.

Reply:
that makes sense! I wasn't quite aware that you could just test a parameter like that at the time. it's quite a lot better than what I had before. Please let me know your thoughts on the revamped version I just pushed